src/cli/add.rs

```diff
@@ -542,12 +605,31 @@ mod tests {
 
     #[test]
     fn test_determine_version_constraint() {
-        insta::assert_snapshot!(determine_version_constraint(&["1.2.0".parse().unwrap()])
+        // Semver
+        insta::assert_snapshot!(determine_version_constraint(&["1.2.0".parse().unwrap(), "1.3.0".parse().unwrap()], PinningStrategy::Semver)
             .unwrap()
-            .to_string(), @">=1.2.0,<1.3");
-
-        insta::assert_snapshot!(determine_version_constraint(&["1.2.0".parse().unwrap(), "1.3.0".parse().unwrap()])
+            .to_string(), @">=1.2.0,<2");
+        insta::assert_snapshot!(determine_version_constraint(&["0.2.0".parse().unwrap(), "0.3.0".parse().unwrap()], PinningStrategy::Semver)
+            .unwrap()
+            .to_string(), @">=0.2.0,<0.4");
+        insta::assert_snapshot!(determine_version_constraint(&["0.2.0".parse().unwrap(), "1.3.0".parse().unwrap()], PinningStrategy::Semver)
+            .unwrap()
+            .to_string(), @">=0.2.0,<2");
+        insta::assert_snapshot!(determine_version_constraint(&["0.2.0".parse().unwrap()], PinningStrategy::Semver)
             .unwrap()
-            .to_string(), @">=1.2.0,<1.4");
+            .to_string(), @">=0.2.0,<0.3");
+
+        // Major
+        insta::assert_snapshot!(determine_version_constraint(&["1.2.0".parse().unwrap(), "1.3.0".parse().unwrap()], PinningStrategy::Major).unwrap().to_string(), @">=1.2.0,<2");
+
+        // Minor
+        insta::assert_snapshot!(determine_version_constraint(&["1.2".parse().unwrap()], PinningStrategy::ExactVersion).unwrap().to_string(), @"==1.2");
+        insta::assert_snapshot!(determine_version_constraint(&["1.2.0".parse().unwrap(), "1.2.0".parse().unwrap(), "1.3.0".parse().unwrap()], PinningStrategy::ExactVersion).unwrap().to_string(), @"==1.2.0|==1.3.0");
+
+        // LatestUP
+        insta::assert_snapshot!(determine_version_constraint(&["1.2.0".parse().unwrap(), "1.3.0".parse().unwrap()], PinningStrategy::LatestUp).unwrap().to_string(), @">=1.2.0");
+
+        // NoPin
+        insta::assert_snapshot!(determine_version_constraint(&["1.2.0".parse().unwrap(), "1.3.0".parse().unwrap()], PinningStrategy::NoPin).unwrap().to_string(), @"*");
     }
 }
```

Comment:
Perhaps it would make sense to create a matrix of two items. A list of a list of versions (or just the min and max), and a list of all possible strategies. Then for each combination compute the spec. Finally create a text output and insta that snapshot. That will give you a nice overview quickly.

Reply:
Changed the test. I believe I did what you asked. And I like it better to!